src/NLog/Layouts/XmlLayout.cs

```diff
@@ -0,0 +1,125 @@
+// SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS 
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN 
+// CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) 
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF 
+// THE POSSIBILITY OF SUCH DAMAGE.
+// 
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Linq;
+using System.Text;
+using System.Xml;
+using NLog.Config;
+using NLog.Internal;
+using NLog.LayoutRenderers;
+
+namespace NLog.Layouts
+{
+    /// <summary>
+    /// An Xml Layout for NLog.
+    /// </summary>
+    [Layout("XmlLayout")]
+    public class XmlLayout : Layout
+    {
+        private readonly IList<XmlProperty> _properties;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="XmlLayout"/> class.
+        /// </summary>
+        public XmlLayout()
+        {
+            _properties = new List<XmlProperty>();
+            Indent = true;
+            OmitXmlDeclaration = true;
+        }
+
+
+        /// <summary>
+        /// Gets the custom properties for the log event.
+        /// </summary>
+        [ArrayParameter(typeof (XmlProperty), "property")]
+        public IList<XmlProperty> Properties
+        {
+            get { return _properties; }
+        }
+
+
+        /// <summary>
+        /// Gets or sets a value indicating whether to indent the XML elements.
+        /// </summary>
+        /// <value>
+        ///   <c>true</c> if indent; otherwise, <c>false</c>.
+        /// </value>
+        [DefaultValue(true)]
+        public bool Indent { get; set; }
+
+        /// <summary>
+        /// Gets or sets a value indicating whether to omit an XML declaration.
+        /// </summary>
+        /// <value>
+        ///   <c>true</c> to omit XML declaration; otherwise, <c>false</c>.
+        /// </value>
+        [DefaultValue(true)]
+        public bool OmitXmlDeclaration { get; set; }
+
+        /// <summary>
+        /// Renders the layout for the specified logging event by invoking layout renderers.
+        /// </summary>
+        /// <param name="logEvent">The logging event.</param>
+        /// <returns>
+        /// The rendered layout.
+        /// </returns>
+        protected override string GetFormattedMessage(LogEventInfo logEvent)
+        {
+            // add custom properties to log event
+            foreach (var xmlProperty in Properties)
+            {
+                string name = xmlProperty.Name;
+                string text = xmlProperty.Layout.Render(logEvent);
+
+                logEvent.Properties[name] =  text;
+            }
+
+            var builder = new StringBuilder();
+            var writer = new LogEventInfoXmlWriter();
+            var settings = new XmlWriterSettings
+            {
+                Indent = Indent,
+                Encoding = Encoding.UTF8,
```

Comment:
Wouldn't it benefit if the Encoding is configurable and Default can be Encoding.UTF8